docs/getting-started/throwing-errors.md

```diff
@@ -0,0 +1,289 @@
+
+Some examples of good human-readable errors:
+
+```js
+throw new Error(`Config file at path ${filePath} does not exist`);
+
+throw new TypeError(`The name parameter must be a string, ${typeof name} was given`);
+
+throw new RangeError(`Age must be between 18 and 137. ${age} was given`);
+```
+
+A common theme in the above is that we're not trying to cater for machines. We're writing instructions which tell an engineer exactly why an error was thrown. We're also including data in the error message which make it non-generic, so it explicitly can't be easily parsed by a machine – that's not the job of the error message.
+
+### Making errors machine readable
+
+It's still important for errors to be machine readable. With the errors above, it's not very easy to group errors in a dashboard or report because the messages are designed for humans. If we have the following error:
+
+```js
+throw new Error(`Config file at path ${filePath} does not exist`);
+```
+
+Then we might see the following errors in our logging dashboard:
+
+```
+Config file at path config/production/default.yml does not exist
+Config file at path mock/config.json does not exist
+```
+
+Great for humans, but bad if we want to run a report like "how many times does the app crash because our config loader fails"? That's where the error `code` property is useful.
+
+Node.js uses `error.code` internally to help us identify error types, e.g.
+
+```js
+try {
+    const example = require('nope');
+} catch (error) {
+    console.log(error.code); // "MODULE_NOT_FOUND" - for machines!
+    console.log(error.message); // "Cannot find module 'nope'" - for humans!
+}
+```
+
+Setting your own `code` property on an built-in error type can be tedious, that's where `OperationalError` comes in:
+
+```js
+// Instead of doing this:
+const error = new Error('A "turnip" is not a valid fruit');
+error.code = 'INVALID_FRUIT';
+throw error;
+
+// You can do this:
+throw new OperationalError({
+    message: 'A "turnip" is not a valid fruit',
+    code: 'INVALID_FRUIT'
+});
+```
+
+### Adding more data
+
+It's also sometimes useful to add extra details to an error so that further debugging of them can be done once we've determined what _type_ of error they are using `code`. Useful information might be the actual value of any input which caused an error, or further details about application state (e.g. the ID of the user this error occurred for).
+
+Taking the example from above where we set the code, we might want to provide the input fruit name alongside our message and code:
+
+```js
+// Assuming the user input is:
+const fruitName = 'turnip';
+
+// You can add any extra data you want here
+throw new OperationalError({
+    message: `A "${fruitName}" is not a valid fruit`,
+    code: 'INVALID_FRUIT',
+    input: fruitName
+});
+```
+
+### Being specific
+
```

Comment:
When should I subclass `OperationalError` and create a system of more specific errors in my application?

Reply:
Yep I don't know the answer to this 😂  I think again let's wait and see what errors the teams find that they need. It may be that if the same error types crop up consistently then they could graduate to the `@dotcom-reliability-kit/errors` package?